File: objective-c/UsingRust/UsingRust_v1/UsingRust_v1/UsingRust_v1/ViewControllers/AMainViewController.m
//
//  AMainViewController.m
//  StoreKit_v2
//
//  Created by Alexandr Kirilov on 15/12/2022.
//

#import "AMainViewController.h"

@implementation AMainViewController

-(instancetype) init {
    
    self = [super init];
    if (self) {

    }
    return self;
}

-(void) loadView {
    
    [super loadView];
    [self.view setBackgroundColor:__COLOR_SALAD_DARK];
    [self setTitle:@"UsingRust_v1"];
}

-(void) viewDidLoad {
 
    [super viewDidLoad];
    
    [self setPButton:[[AButtonTemplate alloc] initWithFrame:CGRectMake(0, 0, 0, 0)]];
    [_pButton setTitle:@"Get from Rust" forState:UIControlStateNormal];
    [_pButton addTarget:self action:@selector(mActionGetDataFromRustLibrary:) forControlEvents:UIControlEventTouchUpInside];
    [self.pViewContent addSubview:_pButton];
    [[_pButton.topAnchor constraintEqualToAnchor:self.pViewContent.topAnchor] setActive:YES];
    [[_pButton.heightAnchor constraintEqualToConstant:80.0] setActive:YES];
    [[_pButton.leftAnchor constraintEqualToAnchor:self.pViewContent.leftAnchor] setActive:YES];
    [[_pButton.rightAnchor constraintEqualToAnchor:self.pViewContent.rightAnchor] setActive:YES];
    
    [self setPLabel:[[ALabelTemplate alloc] initWithFrame:CGRectMake(0, 0, 0, 0 )]];
    [_pLabel setText:@"No data yet"];
    [self.pViewContent addSubview:_pLabel];
    [[_pLabel.centerYAnchor constraintEqualToAnchor:self.pViewContent.centerYAnchor constant:-50.0] setActive:YES];
    [[_pLabel.centerXAnchor constraintEqualToAnchor:self.pViewContent.centerXAnchor] setActive:YES];
    [[_pLabel.leftAnchor constraintEqualToAnchor:self.pViewContent.leftAnchor] setActive:YES];
    [[_pLabel.rightAnchor constraintEqualToAnchor:self.pViewContent.rightAnchor] setActive:YES];
}

-(void) mActionGetDataFromRustLibrary:(id)inSender {
    
    NSLog(@"mActionGetDataFromRustLibrary");
    [_pLabel setText:@(fStringFromRustLibrary())];
}

@end

File: swift/UsingRust/UsingRust_v1/UsingRust_v1/UsingRust_v1/ViewControllers/AMainViewController.swift
//
//  AMainViewController.swift
//  StoreKit2_v1
//
//  Created by Alexandr Kirilov on 19/12/2022.
//

import UIKit

class AMainViewController: AViewControllerTemplate {
	
	var pButton: AButtonTemplate = AButtonTemplate();
    var pLabel: UILabel = UILabel();
	
	override func loadView() {
		
		super.loadView();
		self.title = "UsingRust_v1";
	}
	
	override func viewDidLoad() {
		
		super.viewDidLoad();
        
        pButton.setTitle("Get from Rust", for: .normal);
        pButton.addTarget(self, action: #selector(mActionGetDataFromRustLibrary(inSender:)), for: .touchUpInside);
        pViewContent.addSubview(pButton);
        pButton.topAnchor.constraint(equalTo: pViewContent.topAnchor).isActive = true;
        pButton.heightAnchor.constraint(equalToConstant: 80.0).isActive = true;
        pButton.leftAnchor.constraint(equalTo: pViewContent.leftAnchor).isActive = true;
        pButton.rightAnchor.constraint(equalTo: pViewContent.rightAnchor).isActive = true;
        
        pLabel.text = "No data from Rust";
        pLabel.textColor = __COLOR_WHITE;
        pLabel.font = UIFont.systemFont(ofSize: 20.0);
        pLabel.translatesAutoresizingMaskIntoConstraints = false;
        pLabel.textAlignment = .center;
        pViewContent.addSubview(pLabel);
        pLabel.centerYAnchor.constraint(equalTo: pViewContent.centerYAnchor, constant: -40.0).isActive = true;
        pLabel.heightAnchor.constraint(equalToConstant: 80.0).isActive = true;
        pLabel.leftAnchor.constraint(equalTo: pViewContent.leftAnchor).isActive = true;
        pLabel.rightAnchor.constraint(equalTo: pViewContent.rightAnchor).isActive = true;
	}
    
    // -----------------------------------
    // MARK: Actions
    
    @objc func mActionGetDataFromRustLibrary(inSender:Any) -> Void {
        
        __ALog("mActionGetDataFromRustLibrary");
        
        let oData: String = String(cString: fStringFromRustLibrary());
        pLabel.text = oData;
    }
}


